packages/graphql/src/contracts.js

```diff
@@ -449,6 +455,11 @@ function setMobileBridge() {
     context.marketplace._address
   )
 
+  Object.values(context.marketplaces).forEach(marketplace => {
+    marketplace.contract = context.marketplaceExec
```

Comment:
I'm not familiar at all with this code...
Is it ok to use the same context.marketplaceExec (which is constructed on line 455 above with the same context.marketplace._address) for all the marketplaces ?
If yes, maybe add a comment so that we don't wonder again when we look at this code in the future :)

Reply:
`contract` should be used when making queries whereas `contractExec` should only be used when making a transaction. This is so we only use the wallet's provider when making a transaction and use our own provider otherwise.

It's so we don't just use, eg, Infura for all requests which can have data inconsistencies.

It also looks like this code is using the same marketplace contract for all marketplace versions? That would mean a tx intended for marketplace 000 would go to marketplace 001... not sure what was intended?

Reply:
Maybe we can document this intent in `contracts.js` somewhere.

Reply:
Thanks for these comments. After the refactor each marketplace contract is initialised with the `mobile-bridge` provider  when in right environment and not modified after the initialisation. So this issue is fixed. 

Also added the comments to document the `[contract/web3]Exec` intent as per request of @mikeshultz .